src/ContentClient.ts

```diff
@@ -235,11 +238,63 @@ export class ContentClient implements ContentAPI {
       queryParams.set('fields', [fieldsValue])
     }
 
-    // Reserve a few chars to send the offset
-    const reservedChars = `&offset=`.length + ContentClient.CHARS_LEFT_FOR_OFFSET
+    let reservedChars: number
+    let modifyQueryBasedOnResult: (result: PartialDeploymentHistory<T>, builder: QueryBuilder) => void
+
+    const canWeUseLocalTimestampInsteadOfOffset =
+      deploymentOptions?.fields &&
+      deploymentOptions.fields.isFieldIncluded('auditInfo') &&
+      (!deploymentOptions?.sortBy || deploymentOptions.sortBy.field === SortingField.LOCAL_TIMESTAMP)
+
+    if (canWeUseLocalTimestampInsteadOfOffset) {
+      // Note: the approach used below will get stuck if all 500 deployments on the same page have the same localTimestamp, but that is extremely unlikely
+      reservedChars = '&fromLocalTimestamp='.length + 13
+      if (deploymentOptions?.sortBy?.order === SortingOrder.ASCENDING) {
+        // Ascending
+        modifyQueryBasedOnResult = (result, builder) => {
```

Comment:
This and below are the same, except they are setting a different property in `builder.setParam`. Maybe you can extract to a method, or to an inner function. Maybe you can even use an automatic refactor from VsCode

Reply:
Done!